adapter/internal/synchronizer/apis_fetcher.go

```diff
@@ -234,6 +239,13 @@ func FetchAPIsFromControlPlane(updatedAPIID string, updatedEnvs []string, envToD
 			// Keep the iteration still until all the envrionment response properly.
 			logger.LoggerSync.Errorf("Error occurred while fetching data from control plane for the API %q: %v. Hence retrying..", updatedAPIID, data.Err)
 			sync.RetryFetchingAPIs(c, data, sync.RuntimeArtifactEndpoint, true, queryParamMap)
+			retryCounter++
```

Comment:
Can't we return after satisfying above condition ?

Reply:
I think @slahirucd7  didn't see the break in line 243 (`if data.Resp != nil {` block)